elide-core/src/main/java/com/yahoo/elide/graphql/operations/UpdateOperation.java

```diff
@@ -0,0 +1,216 @@
+                        Set<PersistentResource> newRelationships = new HashSet<>();
+                        if (value != null) {
+                            // TODO: For a to-many relationship this would be a list... Need to handle that case
+                            newRelationships.add(PersistentResourceFetcher
+                                    .load(relName, value.toString(), requestScope));
+                        }
+                        objects.forEach(o -> o.updateRelation(fieldName, newRelationships));
+                    } else if (fieldName.equals(dictionary.getIdFieldName(entityClass))) {
+                        if (id == null) {
+                            // TODO: In case of list, should ensure we're modifying the correct object instance
+                            return; // Not updating id
+                        }
+                        // Set id value
+                        if (value == null) {
+                            throw new WebApplicationException("Cannot set object identifier to null",
+                                    HttpStatus.SC_BAD_REQUEST);
+                        }
+                        objects.forEach(o -> o.setId(value.toString()));
+                    } else {
+                        throw new WebApplicationException("Attempt to update unknown field: '"
+                                + fieldName + "'", HttpStatus.SC_BAD_REQUEST);
+                    }
+                });
+    }
+
+    /**
+     * Check if request expects to replace a root collection.
+     *
+     * If all of the request bodies are applied to a specified id then root is being replaced. However,
+     * if _any_ of the operations apply to the entire dataset as a whole it will be applied in order.
+     *
+     * @return True if replacing root, false otherwise.
+     */
+    private boolean checkIsReplacingRoot() {
+        if (!isRoot) {
+            return false;
+        }
+
+        for (Map<String, Object> input : requestData) {
+            boolean isAppliedToId = false;
+            for (String fieldName : input.keySet()) {
+                if (fieldName.equals(idFieldName)) {
+                    isAppliedToId = true;
+                    break;
+                }
+            }
+            if (!isAppliedToId) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private Collection<PersistentResource> fetchResourceCollection() {
+        if (isRoot) {
+            return new ArrayList<>(fetchRootableResourceCollection());
+        }
+        return new ArrayList<>(fetchNonRootableResourceCollection());
+    }
+
+    private Collection<PersistentResource> fetchNonRootableResourceCollection() {
+        // NOTE: Currently only handles _single_ object update
+        Optional<String> dataId = (id != null) ? Optional.of(id) : field.getArguments().stream()
+                .filter(arg -> ARGUMENT_DATA.equalsIgnoreCase(arg.getName()))
```

Comment:
This code is very similar to some of the code that handles deletes.  I think this kind of logic should move to a Request Specific (Fetch, Delete, Add, Replace) class that provides functions to do common things like get the list of IDs.